tools/cmake/mbed_greentea.cmake

```diff
@@ -3,10 +3,17 @@
 
 option(MBED_TEST_BAREMETAL OFF)
 
-set(MBED_CONFIG_PATH ${CMAKE_CURRENT_BINARY_DIR} CACHE INTERNAL "")
+set(MBED_TARGET_SERIAL_PORT "" CACHE STRING "Serial port of the DUT")
+set(MBED_TARGET_SERIAL_NUM "" CACHE STRING "USB Serial number of the DUT")
+set(MBED_TARGET_MOUNT_POINT "" CACHE STRING "Mount point for the DUT")
 
-include(${CMAKE_CURRENT_LIST_DIR}/app.cmake)
-set(MBED_ROOT ${CMAKE_CURRENT_LIST_DIR}/../.. CACHE INTERNAL "")
+# TODO: After we convert all greentea tests to use CTest, remove this code. We
+# define these parameters in mbed-os/CMakeLists.txt for greentea tests.
+if(NOT BUILD_GREENTEA_TESTS)
+    set(MBED_CONFIG_PATH ${CMAKE_CURRENT_BINARY_DIR} CACHE INTERNAL "")
+    include(${CMAKE_CURRENT_LIST_DIR}/app.cmake)
+    set(MBED_ROOT ${CMAKE_CURRENT_LIST_DIR}/../.. CACHE INTERNAL "")
```

Comment:
Could use `options()` instead of `CACHE`, though it's been like this before this PR.

Reply:
Options only typically allow boolean type ON or OFF values. We need string values for these variables so I think CACHE STRING is more idiomatic.

Reply:
Fair enough